draft-dekater-scion-dataplane.md

```diff
@@ -1342,20 +1355,101 @@ This section describes the steps that a SCION egress border router must perform
+
+
+### On-Path Attacks
+
+In the case of an on-path attack, the attacker sits in the middle of the path between the source endpoint and the destination endpoint and is able to intercept the data packets that are being forwarded. Possible on-path attacks in the data plane are modifications of the SCION path header and SCION address header, thus forcing the packets to travel along paths that may bring the attacker specific advantages. The following subsections provide more details on this kind of on-path data plane attacks.
+
+Also in SCION, an on-path adversary can always simply drop packets. This kind of attack is fundamental and generally cannot be prevented.
+
+**Note:**<br>
+For more information on the SCION path header, see [](#path-header).<br>
+For more information on the SCION address header, see [](#address-header).
+
+
+#### Modification of the Path Header
+
+It is possible for an on-path adversary to modify the SCION path header of a packet, and replace (parts of) path segments with different segments. The only restriction is that the path after the adversary needs to include authorized segments, as otherwise the packet would be simply dropped at some point. But the already traversed portion of the current segment and past segments can be modified by the adversary in a completely arbitrary manner (for instance, deleting and adding valid and invalid hop fields).
+
+On replies by the destination, the adversary can transparently revert the changes to the path header again. For instance, if an adversary M is an intermediate AS on the path of a packet from A to B, then M can replace the packet’s past path (leading up to, but not including M). The new path may not be a valid end-to-end path. However, when B reverses the path and sends a reply packet, that packet would go via M, which can then transparently change the invalid path back to the valid path to A.
+
+The destination endpoint may only discover such path header modifications if some kind of packet integrity protection is in place. In this case, the receiving endpoint verifies that a packet, including its path, is the same as the one sent by the source endpoint. However, SCION currently does not provide this type of protection.
+
+Moreover, packet integrity protection is not enough if there are two colluding adversaries on the path. These colluding adversaries can forward the packet between them using a different path than selected by the source endpoint: The first on-path attacker remodels the packet header arbitrarily, and the second on-path attacker changes the path back to the original source-selected path, such that the integrity check by the destination endpoint succeeds. To prevent this attack and to defend against multiple on-path adversaries in general, path validation is required (which is currently not supported by SCION).
```

Comment:
```suggestion
Moreover, packet integrity protection is not enough if there are two colluding adversaries on the path. These colluding adversaries can forward the packet between them using a different path than selected by the source endpoint: The first on-path attacker remodels the packet header arbitrarily, and the second on-path attacker changes the path back to the original source-selected path, such that the integrity check by the destination endpoint succeeds. To prevent this attack and to defend against multiple on-path adversaries in general, proof of transit is required (which is offered by an experimental extension).
```